crates/uv-python/src/managed.rs

```diff
@@ -504,6 +513,23 @@ impl ManagedPythonInstallation {
         Ok(())
     }
 
+    /// Ensure the environment contains the symlink directory (or junction on Windows)
+    /// pointing to the patch directory for this minor version.
+    pub fn ensure_minor_version_link(&self) -> Result<(), Error> {
+        // We don't currently support transparent upgrades for PyPy or GraalPy
+        // so we don't create a symlink directory (or junction on Windows).
+        if let LenientImplementationName::Known(name) = self.key.implementation() {
+            if matches!(name, ImplementationName::PyPy | ImplementationName::GraalPy) {
+                return Ok(());
+            }
+        }
```

Comment:
This logic seems repeated several times now, is there a clean way for us to abstract it?